### PR TITLE
Port the rest of silkworm's trie algorithm

### PR DESCRIPTION
**Pull request by jomuel**
and apply the existing property test to it

### CODE REVIEW

File: src/trie/intermediate_hashes.rs [hunk 1]
```diff
@@ -282,18 +289,269 @@ where
...
+struct DbTrieLoader<'db, 'tx, 'tmp, 'co, 'nc, Tx>
+where
+    Tx: MutableTransaction<'db>,
+    'db: 'tx,
+    'tmp: 'co,
+    'co: 'nc,
+{
+    txn: &'tx mut Tx,
+    hb: HashBuilder<'nc>,
+    storage_collector: RefCell<&'co mut TableCollector<'tmp, tables::TrieStorage>>,
```
Comment (vorot93) [inline src/trie/intermediate_hashes.rs:307]: Make `calculate_storage_root` accept `&mut self` instead

**Review by vorot93 — APPROVED** [review]
This will obviously need some refactoring. But first let's get it to compute block state roots for us.

File: src/trie/intermediate_hashes.rs (hunk 1 above)
Comment (jomuel) [reply src/trie/intermediate_hashes.rs:307]: That will be part of the refactoring. The `RefCell` here and one or two other hacks were necessary to follow silkworm closely without having two mutable refs at once.

**Comment by jomuel** [comment]
> This will obviously need some refactoring. But first let's get it to compute block state roots for us.

Yes, I'm working on the followup to use it in `stages/interhashes`.

**Pull request merged**